### DISCUSSION TITLE
Welcome to Jsign Discussions!

### DISCUSSION THREAD

**Discussion by ebourg** [ebourg/jsign#155] [Announcements]
## Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together.

If you like this project don't forget to star it!

**Comment by shgchen** [comment]
hi bro~,I encountered a problem when using jar to sign msi files. The jar showed that the signature was successful, but when I checked the file properties, the signature showed 'This digital signature is invalid', and the certificate in the digital signature information showed 'There is no digital signature for the verification object. '.

**Reply by ebourg** [reply]
Did you try with Jsign 5.0? If you can send the unsigned file to ebourg@apache.org I'll get a look (you can use WeTransfer if it's big).

**Reply by shgchen** [reply]
I am using version 5.0 and the msi file is something I randomly found on my computer.

**Reply by ebourg** [reply]
Was it already signed and you added a new signature with Jsign?

**Reply by shgchen** [reply]
It is a file in the computer, it is not signed by jsign, it already has a signature 'Oracle America, Inc.', the file in this location under my computer: C:\Windows\Installer\10e41297.msi

**Reply by shgchen** [reply]
I know what the problem is. It should be the system msi file, which prevents the signature from being modified. I found a test file in the project and it worked. Thank you for your help.

**Reply by ebourg** [reply]
There are some corner cases that aren't properly handled yet, for example when an existing signature is replaced, the new signature may be invalid (with msi files only). But signing an unsigned file should work properly.

**Reply by shgchen** [reply]
OK，thanks,wish u a happy life~~
